perf/bloom_filter/bloom_filter.cc

```diff
@@ -0,0 +1,297 @@
+	 */
+	explicit Tuple(std::uint32_t pk_value)
+	{
+		std::size_t tuple_size =
+			mp_sizeof_array(TUPLE_FIELD_COUNT) +
+			mp_sizeof_uint(pk_value) +
+			mp_sizeof_str(STRING_FIELD_SIZE);
+		begin = (char *)xmalloc(tuple_size);
+		end = mp_encode_array(begin, TUPLE_FIELD_COUNT);
+		end = mp_encode_uint(end, pk_value);
+		end = mp_encode_str(end, (const char *)tuple_string,
+				    STRING_FIELD_SIZE);
+		assert(end == begin + tuple_size);
+
+		std::size_t key_size =
+			mp_sizeof_array(1) +
+			mp_sizeof_uint(pk_value);
+		index_begin = (char *)xmalloc(key_size);
+		index_end = mp_encode_array(index_begin, 1);
+		index_end = mp_encode_uint(index_end, pk_value);
+		assert(index_end == index_begin + key_size);
+	}
+};
+
+/** Represents storage for tuples. */
+struct TupleData {
+	explicit TupleData(std::size_t tuple_count)
+	{
+		selectTuples.reserve(tuple_count);
+		insertTuples.reserve(tuple_count);
+
+		std::cout << "Started filling tupleData vectors." << std::endl;
+		for (std::uint32_t i = 0; i < tuple_count; ++i) {
+			Tuple newTuple(i);
+			insertTuples.push_back(newTuple);
+			selectTuples.push_back(newTuple);
+		}
+		std::cout << "Finished filling tupleData vectors." << std::endl;
+		std::cout << std::endl;
+	}
+
+	TupleData(const TupleData &) = delete;
+
+	TupleData(TupleData &&) = delete;
+
+	~TupleData() {
+		for (auto & t : selectTuples)
+			free(t.begin);
+	}
+
+	std::vector<Tuple> selectTuples;
+	std::vector<Tuple> insertTuples;
+};
+
+/** Tuple that will make select query look up all tree. */
+Tuple *nonexistenetTuple;
+/** Struct that holds:
+ *  1. inserted tuples;
+ *  2. tuples for select queries.
+ */
+TupleData *data;
+
+/** Args passed through lua code. */
+struct BenchArgs {
+	/** Name of benchmarked space. */
+	const char *space_name;
+	/** Space id. */
+	uint32_t space_id;
+	/** Tuple count. */
+	uint32_t tuple_count;
+
+	explicit BenchArgs(const char *args, const char *args_end) {
+		uint32_t arg_count = mp_decode_array(&args);
+		uint32_t name_len = 0;
+		const char *name = mp_decode_str(&args, &name_len);
+		space_name = strndup(name, name_len);
+		space_id = mp_decode_uint(&args);
+		tuple_count = mp_decode_uint(&args);
+	}
+
+	BenchArgs(const BenchArgs &) = delete;
+
+	BenchArgs(BenchArgs &&other) = delete;
+
+	~BenchArgs() { free((void *)space_name); }
+};
+
+/** Trivial wrapper for storing benchmark execution time. */
+struct BenchTimeResult {
+	double elapsed_time;
+};
+
+/** Represents results of benchmark. */
+struct BenchAlgorithmResult {
+	BenchAlgorithmResult(const char *name) : space_name(name) {}
+
+	/** Name of benchmarking space. */
+	const char *space_name;
+	/** Primary index bloom filter binary size. */
+	size_t bloom_bsize = 0;
+	struct BenchTimeResult insertTimeMs;
+	struct BenchTimeResult selectTimeMs;
+};
+
+int
+init(box_function_ctx_t *ctx, const char *args, const char *args_end)
+{
+	uint32_t arg_count = mp_decode_array(&args);
```

Comment:
Please add an assertion that the argument count is correct.

Reply:
Done.